### PR TITLE
Added rest endpoint to fetch space permissions

### PR DESCRIPTION
**Pull request by sushant-glean**

### CODE REVIEW

File: confluence/server/scio_search/src/main/resources/atlassian-plugin.xml [hunk 1]
```diff
@@ -35,4 +35,8 @@
     <description>Provides Scio Search plugin configuration endpoint.</description>
   </rest>
 
+  <rest key="scioSearchSpacePermissionFetch" path="/scio_spaces" version="1.0">
```
Comment (bharath-bhushan-glean) [inline confluence/server/scio_search/src/main/resources/atlassian-plugin.xml]: `/space_permissions` ?

Comment (sushant-glean) [reply confluence/server/scio_search/src/main/resources/atlassian-plugin.xml]: yeah that is better, will update

Comment (sushant-glean) [reply confluence/server/scio_search/src/main/resources/atlassian-plugin.xml]: done

**Comment by sushant-glean** [comment]
example curl request-
```
curl -s --request GET -H 'Authorization: Bearer <ACCESS_TOKEN>' "http://confluence-instance:8090/rest/space_permissions/1.0/?spaceKey=OSP"
```

**Review by bharath-bhushan-glean — APPROVED** [review]

**Review by james-scio — APPROVED** [review]

**Pull request merged**